xmltojsonlegacy/src/main/java/org/assimbly/xmltojsonlegacy/utils/ExtractUtils.java

```diff
@@ -355,9 +359,11 @@ public static void addValueIntoArrayNode(
         switch (type.toLowerCase()) {
             case Constants.JSON_XML_ATTR_TYPE_NUMBER:
                 arrayNode.add(
-                        !config.isElementMustBeNull() ?
-                                subElement!=null ? subElement.asInt() : Integer.parseInt(value) :
-                                null
+                        !config.isElementMustBeNull()
```

Comment:
Two things:

1. Nested tenaries maybe hard to read
2. Is it good to check for double, if it contains a . 

Maybe use a method like this:

public static boolean isDouble(String str) {
    if (str == null || str.isEmpty()) {
        return false;
    }
    try {
        Double.parseDouble(str);
        return true;
    } catch (NumberFormatException e) {
        return false;
    }
}

Reply:
made a few changes regarding your comments.